### PR TITLE
xDai support

### LINKED ISSUE #747: Add xDai support
We should add xDai support to the Conditional Tokens Explorer.

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #747

To test this,

- I have moved some DAI to native xDai with [dai-bridge](https://dai-bridge.poa.network/)  (With 0.01 is enough), there is also a [faucet](https://xdai-faucet.top/)
-  Configure metamask to use xDai in Settings->Networks with `ID: 100`, `RPC:  https://xdai.poanetwork.dev`, `explorer:  https://blockscout.com/poa/xdai`
Captura de pantalla de 2021-01-12 18-04-13
- And to get some ERC20 in xDai [omni-bridge](https://omni.xdaichain.com/), for example DAI to get DAI on xDai

There still some work to be done,
- [x] Configure correctly the CPK to be used in xDai
- [x] Update `.env.example` with the new env vars for xDai
- [x] Add contract for wrapping erc1155 in erc20 to xDai
- [x] Test everything (!). Even filtering because some ERC20 addresses in the graph seem to be the ones in mainnet 

This seems to be working right, the only problem is with the preparation of omen conditions, a change in reality oracle address should solve this.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

File: src/config/networkConfig.ts [hunk 1]
```diff
@@ -272,13 +278,105 @@ const networks: { [K in NetworkIds]: Network } = {
...
+    contracts: {
+      conditionalTokensAddress: CONDITIONAL_TOKEN_CONTRACT_ADDRESS_FOR_XDAI,
+      realityAddress: REALITY_CONTRACT_ADDRESS_FOR_XDAI,
+      wrapped1155FactoryAddress: WRAPPED_1155_FACTORY_CONTRACT_ADDRESS_FOR_XDAI,
+      cpk: {
+        masterCopyAddress: '0x6851D6fDFAfD08c0295C392436245E5bc78B0185',
+        proxyFactoryAddress: '0xfC7577774887aAE7bAcdf0Fc8ce041DA0b3200f7',
+        multiSendAddress: '0x035000FC773f4a0e39FcdeD08A46aBBDBF196fd3',
+        fallbackHandlerAddress: '0x602DF5F404f86469459D5e604CDa43A2cdFb7580',
+      },
```
Comment (hpmaxi) [inline src/config/networkConfig.ts:292]: This should make the cpk works in xDai. But more research is needed.

File: src/config/networkConfig.ts [hunk 2]
```diff
@@ -272,13 +278,105 @@ const networks: { [K in NetworkIds]: Network } = {
...
+      {
+        symbol: 'USDC',
+        address: '0xddafbb505ad214d7b80b1f830fccc89b60fb7a83',
+        decimals: 6,
+      },
+      // {
+      //   symbol: 'CDAI',
+      //   address: '0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643',
+      //   decimals: 18,
+      // },
```
Comment (hpmaxi) [inline src/config/networkConfig.ts:309]: I couldn't find some tokens in xDai. Left commented out to add them later.

**Comment by elena-zh** [comment]
Hey @hpmaxi !
I have taken a look into the PR . Cool, now I' able to open and navigate in the app in the xDai network, however, you know, there are a lot of issue exist and a lot of features are not implemented yet. Anyway, here is my feedback:

1. Filter by 'REPORTER / ORACLE' = Reality.eth does not return any values. However, look: there are 417 conditions, 398 are filtered by 'REPORTER / ORACLE' = Custom Reporter. And the rest 19 conditions are missing (I assume, they are Omen conditions)



2. When open a question retails from a condition, it appears that Reality.eth does not supports xDai. Looks a bit strange for me.  I know, that we do not fix issues for reality.eth, but, I think, we should come up with a decision how to manage this case in our app.



3. Positions list and details do not show collateral names. Filtering by collateral does not work as well


4. I was able to create a custom reporter condition. However, I still cannot see in in the conditions list. May be, it will appear later... 
(id = 0x9b7baf04ede938d53b1e2567138ba0933eac825adfd2a6f4f2f0f633461405d5)

5. I was not able to create Omen condition and split a position. Each time I get an error.. =(



That it from my side so far. Due to I'm blocked in testing by the case 5.

**Comment by hpmaxi** [comment]
Thanks @elena-zh !!!
Most of those things are related to the last point of the graph not using the correct addresses. But I'll re-check one by one when that is solved.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by davidalbela** [comment]
The fixed xDai subgraph to check this PR was deployed here: https://thegraph.com/explorer/subgraph/davidalbela/conditional-tokens-xdai?selected=playground

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

File: src/config/networkConfig.ts (hunk 1 above)
Comment (davidalbela) [reply src/config/networkConfig.ts:308]: Let's take a look https://github.com/gnosis/contract-proxy-kit/blob/development/src/config/networks.ts#L73

**Comment by elena-zh** [comment]
Hi @hpmaxi !
I have re-run tests. So, the app works much-much better now in xDai. However, several issue remain to be unfixed by now: 
1. Impossible to create an Omen condition
2. Impossible to run transaction when the 'Use CPK Address' option is switched on
3. wrapped collateral details in Etherscan do not show any transactions.. https://rinkeby.etherscan.io/token/0x7faf961c917828adaace6562707ebf0788c1f1c6
I'm not sure if it is a case that we should handle, but anyways, that's strange. 
4. and still Reality.eth does not supports xDai, hence, it is impossible to see question details for existing Omen conditions.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by hpmaxi** [comment]
> Hi @hpmaxi !
> I have re-run tests. So, the app works much-much better now in xDai. However, several issue remain to be unfixed by now:
> 
> 1. Impossible to create an Omen condition
> 2. Impossible to run transaction when the 'Use CPK Address' option is switched on
> 3. wrapped collateral details in Etherscan do not show any transactions.. https://rinkeby.etherscan.io/token/0x7faf961c917828adaace6562707ebf0788c1f1c6
>    I'm not sure if it is a case that we should handle, but anyways, that's strange.
> 4. and still Reality.eth does not supports xDai, hence, it is impossible to see question details for existing Omen conditions.

@elena-zh 
1 and 2 should be working now.
3. https://blockscout.com/poa/xdai/address/0x7FAf961c917828AdAAce6562707EbF0788C1F1C6/transactions I guess the problem is that etherscan doesn't seem to support xDai. 
4. That's right.

**Comment by elena-zh** [comment]
HI @hpmaxi !

I have re-tested the changes. I was able to create an Omen condition now. However, here are some issues:
1.  all the transactions fail when I switch on the 'Use CPK Address' option. I have created a separate issue for this case - #816.
2. Then, I was not able to connect to the app with xDAI network using WalletConnect. Here the issue #815
3. And an additional question: do we need to list the collateral tokens in the positions list filter that are not used in xDAI? I mean, USDC or something like this.. However, I might be wrong there as I am not aware of what tokens may be used in xDAI network...

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by hpmaxi** [comment]
> 3. And an additional question: do we need to list the collateral tokens in the positions list filter that are not used in xDAI? I mean, USDC or something like this.. However, I might be wrong there as I am not aware of what tokens may be used in xDAI network...

I think we shouldn't list something that is not inside the network we are connected at the moment but is possible that some tokens must be added to the xDAI configuration. I just sent support for WXDAI. Thanks!

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by elena-zh** [comment]
Hi Maxi! I have taken a look into the changes. I was able to connect the app using wallet connect. 
And thanks for adding WXDAI to the collateral list. 
I still cannot run transactions using CPK address, however, the root cause of the issue might be that I do not have enough xDAI in my wallet to pay for transaction.

**Review by davidalbela — APPROVED** [review]
LGTM

The issue with Walletconect + CPK needs to be fixed on other PR, I think this is other feature to include in the future meanwhile this PR is a feature for xDai support.

What do you think? :)

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr793--conditionaltokensexplorer.review.gnosisdev.com

**Pull request merged**